go/logic/orchestrator.go

```diff
@@ -142,16 +166,43 @@ func discoverInstance(instanceKey inst.InstanceKey) {
 	discoveriesCounter.Inc(1)
 
 	// First we've ever heard of this instance. Continue investigation:
-	instance, err = inst.ReadTopologyInstanceBufferable(&instanceKey, config.Config.BufferInstanceWrites)
+	instance, err = inst.ReadTopologyInstanceBufferable(&instanceKey, config.Config.BufferInstanceWrites, latency)
 	// panic can occur (IO stuff). Therefore it may happen
 	// that instance is nil. Check it.
 	if instance == nil {
 		failedDiscoveriesCounter.Inc(1)
-		log.Warningf("discoverInstance(%+v) instance is nil in %.3fs, error=%+v", instanceKey, time.Since(start).Seconds(), err)
+		discoveryMetrics.Append(&discovery.Metric{
+			Timestamp:       time.Now(),
+			InstanceKey:     instanceKey,
+			TotalLatency:    latency.Elapsed("total"),
+			BackendLatency:  latency.Elapsed("backend"),
+			InstanceLatency: latency.Elapsed("instance"),
+			Err:             err,
+		})
+		log.Warningf("discoverInstance(%+v) instance is nil in %.3fs (Backend: %.3fs, Instance: %.3fs), error=%+v",
+			instanceKey,
+			latency.ElapsedSeconds("total"),
+			latency.ElapsedSeconds("backend"),
+			latency.ElapsedSeconds("instance"),
+			err)
```

Comment:
Is it possible that between the `discoveryMetrics.Append(...)` call and this line, the returned value of `ElapsedSeconds(...)` are incompatible? Would this be confusing?